File: sj/75week/Sum of All Odd Length Subarrays.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var sumOddLengthSubarrays = function(arr) {
    
    let sum = 0;
    
    for(let i=0; i<arr.length; i++)
    {
        for(let j=i; j<arr.length; j++)
            {
                if((i-j)%2==0)
                    {
                        for(let k=i; k<=j; k++)
                            {
                                sum+= arr[k];
                            }
                    }
            }
    }
    
    return sum;
};

File: jw/week75/SumOfAllOddLengthSubarrays.java
class Solution {

    public int sumOddLengthSubarrays(int[] arr) {
        int sum = 0;
        for (int i = 0; i < arr.length; i += 1) {
            for (int j = i + 1; j <= arr.length; j += 2) {
                int[] subArray = Arrays.copyOfRange(arr, i, j);
                for (int k = 0; k < subArray.length; k += 1) {
                    sum += subArray[k];
                }
            }
        }
        return sum;
    }
   
}
